inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/DateExtractFunction.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.sdk.transform.process.function;
+
+import org.apache.inlong.sdk.transform.decode.SourceData;
+import org.apache.inlong.sdk.transform.process.Context;
+import org.apache.inlong.sdk.transform.process.operator.OperatorTools;
+import org.apache.inlong.sdk.transform.process.parser.ValueParser;
+
+import net.sf.jsqlparser.expression.Expression;
+import net.sf.jsqlparser.expression.Function;
+
+import java.sql.Date;
+import java.time.LocalDate;
+import java.time.temporal.TemporalField;
+import java.time.temporal.WeekFields;
+import java.util.List;
+import java.util.Locale;
+
+/**
+ * DateExtractFunction
+ * description:
+ * - year(date)--returns the year from SQL date date
+ * - quarter(date)--returns the quarter of a year (an integer between 1 and 4) from SQL date date
+ * - month(date)--returns the month of a year (an integer between 1 and 12) from SQL date date
+ * - week(date)--returns the week of a year (an integer between 1 and 53) from SQL date date
+ * - dayofyear(date)--returns the day of a year (an integer between 1 and 366) from SQL date date
+ * - dayofmonth(date)--returns the day of a month (an integer between 1 and 31) from SQL date date
+ */
+public class DateExtractFunction implements ValueParser {
+
+    private int type;
+    private ValueParser dateParser;
+    private static final TemporalField weekOfYearField = WeekFields.of(Locale.getDefault()).weekOfWeekBasedYear();
+
+    /**
+     * Constructor
+     *
+     * @param type the type of date extract function(1->year, 2->quarter, 3->month, 4->week, 5->dayofyear, 6->dayofmonth)
+     * @param expr
+     */
+    public DateExtractFunction(int type, Function expr) {
+        this.type = type;
+        List<Expression> expressions = expr.getParameters().getExpressions();
+        dateParser = OperatorTools.buildParser(expressions.get(0));
+    }
```

Comment:
Why not use meaningful words to represent the type like "YEAR", "MONTH"

Reply:
fixed.